SpatialGDK/Source/SpatialGDK/Public/Utils/RPCContainer.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/RPCPayload.h"
+#include "Schema/UnrealObjectRef.h"
+#include "SpatialConstants.h"
+
+#include "UObject/WeakObjectPtr.h"
+#include "CoreMinimal.h"
+
+struct FUnrealObjectRef;
+struct FPendingRPCParams;
+using FPendingRPCParamsPtr = TUniquePtr<FPendingRPCParams>;
```

Comment:
I'm still not sure this needs to be a pointer. This seems like a premature optimisation that I'm not sure is actually an optimisation. It's not that big a deal, but I think making the unique pointer means we do the same number of moves of the internal data as if we just move the base type around.

Reply:
I'm not sure either. I've created a jira https://improbableio.atlassian.net/browse/UNR-1650 to profile and optimise RPCContainer code.

Reply:
I'd rather not make specific tickets for things like this unless we know they are slow. I'm suggesting we don't introduce extra layers of indirection without reason though.